pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/CalmWorker.scala

```diff
@@ -0,0 +1,57 @@
+package uk.ac.wellcome.platform.transformer.calm
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.amazonaws.services.sqs.model.Message
+import uk.ac.wellcome.bigmessaging.BigMessageSender
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.models.work.internal.TransformedBaseWork
+import uk.ac.wellcome.platform.transformer.calm.models.CalmRecord
+import uk.ac.wellcome.storage.store.VersionedStore
+import uk.ac.wellcome.storage.{Identified, Version}
+
+sealed trait CalmWorkerError extends Throwable
+object DecodeKeyError extends CalmWorkerError
+object StoreReadError extends CalmWorkerError
+object TransformerError extends CalmWorkerError
+object MessageSendError extends CalmWorkerError
```

Comment:
I think we probably want some more info about these errors rather than just a type. Could something like this work instead?

```scala
sealed class CalmWorkerError(msg: String) extends Exception(msg)
class DecodeKeyError(msg: String) extends CalmWorkerError(msg)
class StoreReadError(msg: String) extends CalmWorkerError(msg)
...
```

On a side note, I think we should avoid using the `Throwable` type to represent application level exceptions, and instead use `Exception` (which is what that class is generally for in Java)

Reply:
Definitely do - I was going to add that when implementing the flow on the error sink. Happy to tackle that as a next PR.